### ISSUE TITLE
评价指标问题

### ISSUE THREAD

**Issue by NonCover** [tongnie/ImputeFormer#5]
作者你好！在你的实验中，在进行训练时，你计算损失函数使用的是`loss = self.loss_fn(imputation, y_loss, mask)`，这里为什么不用eval_mask。
感谢！

**Comment by tongnie** [comment]
您好！在模型训练中mask为在训练过程中给训练数据添加的mask，用于计算训练集loss; eval_mask用于计算验证集loss, 例如：
https://github.com/tongnie/ImputeFormer/blob/5bc7f6383624cbdf4a295d44456dcc9baed41bec/experiments/imputeformer/imputers/imputeformer_imputer.py#L112

**Comment by NonCover** [comment]
谢谢您百忙之中的回复，由于我是这个领域的初学者，有许多问题不是太明白。
我的理解是：training_mask（0代表进行掩码）是用于给数据进行掩码，eval_mask（1代表需要计算的值）是用于进行计算损失函数的。但是如果将mask用于训练集的损失计算的话，计算出来的值并不是缺失部分的损失值，而是非缺失部分？

**Comment by tongnie** [comment]
您好，training_mask的计算如下：
https://github.com/tongnie/ImputeFormer/blob/5bc7f6383624cbdf4a295d44456dcc9baed41bec/experiments/imputeformer/datasets/prototypes/mixin.py#L223

在进行训练时，每个batch的数据会被额外进行mask用于计算训练loss，即以下在原始数据mask基础上的whiten过程：
![image](https://github.com/user-attachments/assets/1578b5cb-c354-4ac5-a2d0-35a35969dbe7)

在计算训练loss时使用的mask为injected_missing：
https://github.com/tongnie/ImputeFormer/blob/5bc7f6383624cbdf4a295d44456dcc9baed41bec/experiments/imputeformer/imputers/spin_imputer.py#L54

因此计算训练集loss时是在人工mask掉的部分进行，希望对您有帮助！

**Comment by NonCover** [comment]
感谢作者大佬，大致了解的整体的思路。代码中的batch.mask表示的是training_mask是吧

**Comment by tongnie** [comment]
嗯嗯是的，传给loss计算的mask即training_mask
